### PR TITLE
Enable & fix cascade_invocations

### LINKED ISSUE #77438: Enable cascade_invocations lint
1. Comment in this line: https://github.com/flutter/flutter/blob/c9d68ba75f73155ca3882e3a17657022f0ac1a89/analysis_options.yaml#L98
2. Run `flutter analyze --flutter-repo --watch`
3. Fix all warnings printed by previous command
4. Create PR

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
This enables the `cascade_invocations` lint and fixes all warnings produced by that.

Fixes #77438.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by creativecreatorormaybenot** [comment]
@Hixie test-exempt? :)

Also, how do we review a PR like this? It would be ~6k LOC changed once all issues are fixed 🤔

**Comment by creativecreatorormaybenot** [comment]
Note the fixes in this PR are only completed by 1/4. I am waiting on @goderbauer's input before continuing here (see https://github.com/flutter/flutter/issues/77438#issuecomment-904952242).

File: packages/flutter/lib/src/cupertino/button.dart [hunk 1]
```diff
@@ -212,13 +212,13 @@ class _CupertinoButtonState extends State<CupertinoButton> with SingleTickerProv
...
-    final TickerFuture ticker = _buttonHeldDown
+    _buttonHeldDown
         ? _animationController.animateTo(1.0, duration: kFadeOutDuration, curve: Curves.easeInOutCubicEmphasized)
-        : _animationController.animateTo(0.0, duration: kFadeInDuration, curve: Curves.easeOutCubic);
-    ticker.then<void>((void value) {
-      if (mounted && wasHeldDown != _buttonHeldDown)
-        _animate();
-    });
+        : _animationController.animateTo(0.0, duration: kFadeInDuration, curve: Curves.easeOutCubic)
+        .then<void>((void value) {
```
Comment (nt4f04uNd) [inline packages/flutter/lib/src/cupertino/button.dart:218]: I might be wrong, but it seems this `then` now only works on the `:`, but not `?`. In my opinion this is a good example of why refactoring that much of code just to meed another style might not worth it

Comment (creativecreatorormaybenot) [reply packages/flutter/lib/src/cupertino/button.dart:218]: @nt4f04uNd you are wrong (not meant negatively, just to answer the "I might be wrong" part 😄).

Comment (creativecreatorormaybenot) [reply packages/flutter/lib/src/cupertino/button.dart:218]: Although I did not go into the full snippet here - I am pretty sure this is what was changed in the latest stable release regarding cascade formatting as well (to make this clear). But I might be wrong 😄

Comment (nt4f04uNd) [reply packages/flutter/lib/src/cupertino/button.dart:218]:
```dart
void main() {
  test(true);
  test(false);
  test2(true);
  test2(false);
}

void test(bool value) {
  final future = value ? Future.value('foo') : Future.value('bar');
  future.then((string) { print(string); });
}

void test2(bool value) {
  value
    ? Future.value('foo')
    : Future.value('bar')
  .then((string) { print(string); });
}
```

prints out

```
foo
bar
bar
```

**Comment by Hixie** [comment]
Looks like we dropped the ball on reviewing this.

@goderbauer Can you take a look and see if we want to enable this lint?

**Comment by creativecreatorormaybenot** [comment]
To be honest, the code base currently does not adhere to it one bit, so I am not sure how much sense it makes to enable it.

**Comment by goderbauer** [comment]
Does `dart fix` support auto-fixing this lint?

There are a lot of cases that look cleaner, but many also do look strange. Maybe we shouldn't apply this lint at this time.

**Comment by creativecreatorormaybenot** [comment]
@goderbauer I am not sure - the only way to find out would be running it, right?

**Comment by goderbauer** [comment]
That would be one way. There may also be a list somewhere.

**Comment by creativecreatorormaybenot** [comment]
I saw that you changed the comment to say that the lint does not match the repo. In that case, I will close the PR.

**Pull request closed without merging**